src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const clonedState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clonedState, action.extraData);
+        result.push(Object.assign({}, clonedState));
+        break;
+
+      case 'removeProperties':
+        const arr = action.keysToRemove;
+
+        for (const element of arr) {
+          delete clonedState[element];
+        }
+
+        result.push(Object.assign({}, clonedState));
+        break;
+
+      case 'clear':
+        for (const key in clonedState) {
+          if (clonedState.hasOwnProperty(key)) {
+            delete clonedState[key];
+          }
+        }
+
+        result.push(Object.assign({}, clonedState));
+        break;
+    }
```

Comment:
You have `result.push` on each case. You can push after switch statement
```suggestion
    switch (action.type) {
      case 'addProperties':
        Object.assign(clonedState, action.extraData);
        
        break;

      case 'removeProperties':
        const arr = action.keysToRemove;

        for (const element of arr) {
          delete clonedState[element];
        }

        break;

      case 'clear':
        for (const key in clonedState) {
          if (clonedState.hasOwnProperty(key)) {
            delete clonedState[key];
          }
        }

        break;
    }
    result.push({...clonedState});
```

Reply:
done